pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -886,6 +898,95 @@ func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMe
 	return err
 }
 
+// isActionTimedOut checks if time since timestamp stored in specified machine annotation did not exceed given timeout
+func isActionTimedOut(machine *machinev1beta1.Machine, annotation string, timeout time.Duration) bool {
+	annotations := machine.Annotations
+	if annotations == nil {
+		return false
+	}
+	startTimeString, exist := annotations[annotation]
+	if !exist {
+		log.Printf("Annotation %q not found on machine %q", annotation, machine.Name)
```

Comment:
Is this an error? A failure? Informational? Can the user do anything about it? Should they?

Reply:
Well, it should inform the user about a non critical failure (with basically no impact as the remediation process will be the same as before) and there is nothing user can do about it. Do you think that we should not log anything in this case?

One improvement that comes to my mind is to try to update the annotation with the current time which will ensure that there is always a valid timestamp (the fact that it was added later should not be a big deal). In case there is really an issue with powering machine back on, we will be able to reprovision the machine eventually.

Reply:
I wouldn't log the message unless we're going to use it for debugging, and in that case I would want the message to say how it is handling the missing data. Maybe `Annotation %q not found on machine %q, assuming remediation has not timed out`?

Reply:
Sounds good, fixed.